### PR TITLE
enableLocationPermission return type changed to Future<bool>

### LINKED ISSUE #9: Return Future<bool> in askLocationPermission
Change the return type of askLocationPermission function from `void` to `Future<bool>`.
Return `true` if the user grants permission, `false` otherwise.

### PR DESCRIPTION
**Pull request by GouravSaini125**
enableLocationPermission return type changed to Future<bool>.
This fixes #9

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 2.0.1-dev
```
Comment (mannprerak2) [inline CHANGELOG.md:1]: Please change the version to 2.0.0, So that I can publish it now.

Comment (mannprerak2) [reply CHANGELOG.md:1]: You can merge the change log of `2.0.0-dev` and `2.0.1-dev` into `2.0.0`

Comment (mannprerak2) [reply CHANGELOG.md:1]: Never mind, I will update it

**Comment by mannprerak2** [comment]
Thank you @GouravSaini125 🎊 🎊 🎊

**Comment by GouravSaini125** [comment]
Thank you too :smile: :smile: :smile:

**Comment by mannprerak2** [comment]
@GouravSaini125 
There is some issue in building the project after your PR can you look into it?
Although the example is working, new projects are having some issues.
To reproduce the issue -
1. Make a new empty project
2. Add nearby_connections: ^2.0.0
3. Run flutter build apk

**Comment by mannprerak2** [comment]
Never mind, this was a trivial issue, I have fixed it.

**Comment by GouravSaini125** [comment]
@mannprerak2 I'm Sorry, can u please explain why it was not throwing error for empty constructor earlier. 
And how can i connect with you...

**Comment by mannprerak2** [comment]
@GouravSaini125 
It is mentioned here https://flutter.dev/docs/development/packages-and-plugins/plugin-api-migration
upgrade steps -> 4
https://www.linkedin.com/in/prerak-mann-4b0011165

**Comment by GouravSaini125** [comment]
@mannprerak2 Ohh man, thnx, I see :rofl::rofl:
Already have sent conn req earlier.... :hugs:

**Pull request merged**